src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs

```diff
@@ -202,11 +202,11 @@ public static IntPtr GetDelegateLdFtnResult(Delegate d, out RuntimeTypeHandle ty
             return d.GetFunctionPointer(out typeOfFirstParameterIfInstanceDelegate, out isOpenResolver, out isInterpreterEntrypoint);
         }
 
-        public static void GetDelegateData(Delegate delegateObj, out object firstParameter, out object helperObject, out IntPtr extraFunctionPointerOrData, out IntPtr functionPointer)
+        public static unsafe void GetDelegateData(Delegate delegateObj, out object firstParameter, out object helperObject, out IntPtr extraFunctionPointerOrData, out IntPtr functionPointer)
         {
             firstParameter = delegateObj.m_firstParameter;
             helperObject = delegateObj.m_helperObject;
-            extraFunctionPointerOrData = delegateObj.m_extraFunctionPointerOrData;
+            extraFunctionPointerOrData = (IntPtr)delegateObj.m_extraFunctionPointerOrData;
```

Comment:
Why do we need the `(IntPtr)` cast here? I thought that `nint` is implicitly castable to `IntPtr`.

Reply:
I was trying to convert `m_extraFunctionPointerOrData` to `void*`, but reverted that change because it needs a lot more change like this one, so reverted that, this is left over, will be removed